modules/utils/include/hephaestus/utils/signal_handler.h

```diff
@@ -0,0 +1,44 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <atomic>
+#include <csignal>
+
+#include <range/v3/range/conversion.hpp>
+
+namespace heph::utils {
+
+/// \brief Use this class to block until a signal is received.
+/// > NOTE: can be extended to call a generic callback when a signal is received.
+/// Usage:
+/// ```
+/// int main() {
+///  // Do something
+///  SignalHandlerStop::wait();
+/// }
+/// // Or
+/// while (SignalHandlerStop::ok()) {
+/// // Do something
+/// }
+/// ```
+class SignalHandlerStop {
+public:
+  /// Return false if a signal has been received, true otherwise.
+  [[nodiscard]] static auto ok() -> bool;
```

Comment:
```suggestion
  [[nodiscard]] static auto stopRequested() -> bool;
```
with inverted logic

Reply:
you are right, I was following ROS logic...fool of me 🤦